crates/wasmparser/src/readers/core/tables.rs

```diff
@@ -74,17 +74,28 @@ impl<'a> FromReader<'a> for TableType {
         let has_max = (flags & 0b001) != 0;
         let shared = (flags & 0b010) != 0;
         let table64 = (flags & 0b100) != 0;
+
+        if shared && !reader.shared_everything_threads() {
+            bail!(
+                pos,
+                "shared tables require the shared-everything-threads proposal"
+            );
+        }
+        if table64 && !reader.memory64() {
+            bail!(pos, "memory64 must be enabled for 64-bit tables");
+        }
```

Comment:
Similar to above with memories, can these be moved back to the validator and the local tests changed to `assert_invalid`?